### ISSUE TITLE
continuous_update on IntText requires mouseout before updating

### ISSUE THREAD

**Issue by andymcarter** [jupyter-widgets/ipywidgets#1753]
I was having difficulty figuring out why widgets were slow to interact with a changing IntText when modified by the up and down arrows. With continuous_update set to False (default), the IntText actually requires a mouseout (i.e. move the cursor off the widget button) before the observed function occurs.

Is this intended behaviour? I would have expected the function to fire once I had stopped depressing the mouse key if I wanted to change the IntText by a large amount at once e.g. increase from 10 to 30, and see the reaction as soon as I stop holding the mouse key when the text hits 30.

**Comment by pbugnion** [comment]
With `continuous_update` set to `False`, the view only sends update to the model when the browser sends the `change` event. With `continuous_update` set to `True`, the view sends updates to the model in response to both `change` and `input` events. The difference between these is described thoroughly [here](https://developer.mozilla.org/en-US/docs/Web/Events/change).

When @jasongrout  implemented `continuous_update`, he rationalized defaulting to `False` for text boxes as (PR #1545):

> I default continuous_update to False for the textboxes because I figured if people were typing, that's typically what is wanted.

That makes sense for inputs which are purely text, but maybe it makes less sense for inputs which also allow mouse interaction, like `{Float,Int}Text`.

**Comment by pbugnion** [comment]
> What browsers do this?

Firefox and Safari seem to do the 'right thing' -- responding to mouse clicks by sending *change* events at least when the user depresses the mouse. Chrome (61) is the offender here -- it only seems to send the change event when the user's mouse moves away from the input, or on blur.

**Comment by andymcarter** [comment]
Yes I was using Chrome, thanks for the in-depth explanation. I am hooking up some bqplot charts to widgets, just running dummy functions at the moment so I might switch back to false when I have more expensive functions in place.

**Comment by jasongrout** [comment]
@pbugnion - do you know if there is a bug report for Chrome?

**Comment by pbugnion** [comment]
A brief look hasn't revealed anything. I'll spend more time checking this tomorrow, and, if not, I can try submitting one.

Here's a minimal HTML snippet that demonstrates the behaviour.

```html
<html>
    <head>

        <script>
         window.onload = () => {
             var element = document.getElementById('example-input');
             element.onchange = () => console.log('change');
             element.oninput = () => console.log('input');
         }
        </script>

    </head>
    <body>
        <input id="example-input" type="number" />
    </body>
</html>
```

**Comment by jasongrout** [comment]
Possibly relevant: https://bugs.jquery.com/ticket/8169

**Comment by pbugnion** [comment]
I found [this bug report](https://bugs.chromium.org/p/chromium/issues/detail?id=748073&q=input%20type%20number&colspec=ID%20Pri%20M%20Stars%20ReleaseBlock%20Component%20Status%20Owner%20Summary%20OS%20Modified) for Chromium.

The developers seem to agree that it's a bug that was introduced in Chrome 58. The design would be to dispatch change on `<mouse-up>`, which seems similar to what Firefox does.

I would personally be in favour of just waiting it out until the Chrome bug is fixed. Maybe we could add a note in the documentation?

**Comment by jasongrout** [comment]
I updated the title to note that we should document this bug of Chrome
